Homework11/Homework11/Program.cs

```diff
@@ -1,15 +1,52 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.Text.Json;
 
 namespace Homework11
 {
     internal class Program
     {
-        static void Main(string[] args)
+        public static void Main(string[] args)
         {
+            try
+            {
+                Catalog catalog = new Catalog();
+
+                Product newProduct1 = new Product("Product1", 15, 10);
+                Product newProduct2 = new Product("Product2", 30, 20);
+                Product newProduct3 = new Product("Product3", 45, 30);
+
+                catalog.AddProducts(newProduct1);
+                catalog.AddProducts(newProduct2);
+                catalog.AddProducts(newProduct3);
+
+                Console.WriteLine("Serialized Products:");
+
+                List<string> serializedProducts = catalog.SerializeJsonDemo();
+
+                string newProduct4 = "[{\"Name\":\"Product4\",\"Price\":60,\"Quantity\":40}]";
+
+                List<string> serializedProducts2 = new List<string> { newProduct4 };
+
+                Console.WriteLine("Deserialized Products:");
+
+                catalog.DeserializeJsonDemo(serializedProducts2);
+                
+                foreach (var product in catalog.Products)
+                {
+                    Console.WriteLine($"Name: {product.Name}, Price: {product.Price}, Quantity: {product.Quantity}");
+                }
+
+            }
+            catch (InvalidPriceException ex)
+            {
+                Console.WriteLine($"Error: {ex.Message}");
+            }
+            catch (JsonException ex)
+            {
+                Console.WriteLine($"Error: {ex.Message}");
+            }
```

Comment:
3 veci:
1. pokud mas stejnou logiku, kterou chces provest po chyceni 2 ruznych typu vyjimek, tak se da vyhnout duplicite
2. Tady jsem na vrchu aplikace, vyplatilo by se tady mit i obecny `catch (Exception ex)` statement
3. napr. JsonException handlujes uz v ramci `catalog.DeserializeJsonDemo()`, takze sem do toho catch bloku se execution nikdy nedostane. V praxi validni pouziti je, ze napr. v te metode vyjimku chytis, provedes nejakou logiku (v tvem pripade udelas treba vypis) a vyhodis tu vyjimku znovu `throw ex;`, pak by se ti dostala az sem do tohohle catch bloku :) Sdilim jen pro info, abys vedel, jak se vyjimky muzou zpracovavat na vicero urovnich

Reply:
1. by se dala implementovat napr. takto
`catch (Exception e) when (e is JsonException || e is InvalidPriceException) {...}`